packages/ec-core/src/operator/composable/repeat_with.rs

```diff
@@ -23,21 +22,28 @@ impl<F, Input, const N: usize> Operator<Input> for RepeatWith<F, N>
 where
     Input: Clone,
     F: Operator<Input>,
+    anyhow::Error: From<F::Error>,
 {
     type Output = [F::Output; N];
-
-    fn apply(&self, input: Input, rng: &mut rand::rngs::ThreadRng) -> Result<Self::Output> {
-        iter::repeat_with(|| self.f.apply(input.clone(), rng))
+    type Error = anyhow::Error;
+
+    fn apply(
+        &self,
+        input: Input,
+        rng: &mut rand::rngs::ThreadRng,
+    ) -> Result<Self::Output, Self::Error> {
+        #[allow(clippy::panic)]
+        Ok(iter::repeat_with(|| self.f.apply(input.clone(), rng))
             .take(N)
-            .try_collect::<_, Vec<<F as Operator<Input>>::Output>, anyhow::Error>()?
+            .try_collect::<_, Vec<_>, _>()?
```

Comment:
Tbh I would just use 
```suggestion
            .collect::<Result<Vec<_>, _>>()?
```
here, then we don't have the additional dependency on itertools (per itertools docs is try_collect just a shorthand for exactly that) but if you prefer this we can keep it this way.

Reply:
I agree with you here. No point in bring in `try_collect` from `itertools` when `.collect()` will work fine. My guess is that when this code was written over a year ago, when I didn't really understand the ability to directly collect `Result<Vec<_>, _>`. I probably thought I "needed" `try_collect` here. There's one other use of `try_collect()` (in `generation.rs`), but I do think that's more readable as is, in part because the compiler is able to fully infer the types there with `try_collect()`, where we have to specific types if we use `collect()`.